core/src/test/java/org/apache/calcite/util/graph/DirectedGraphTest.java

```diff
@@ -64,14 +64,21 @@
     g.addEdge("B", "D");
     assertEquals("[A, B, D]", shortestPath(g, "A", "D").toString());
     assertNull(shortestPath(g, "A", "E"), "There is no path from A to E");
-    assertEquals("[]", shortestPath(g, "D", "D").toString());
+    assertEquals("[D, C, D]", shortestPath(g, "D", "D").toString());
```

Comment:
This seems a regression, I think we should put somewhere (`getPaths`? `findPaths`?) that `if (from.equals(to))` the empty list is a valid path (as it happened previously in the old `getShortestPath` method).

Reply:
Sounds reasonable. I have revised the code to include the empty path to itself.

BTW, the empty path from node D to itself should be denoted "[D]"? as it helps to maintain the relation:

pathLength == path.size() - 1

Reply:
Ok.